### ISSUE TITLE
Add ` toch.no_grad()` for feature computation

### ISSUE THREAD

**Issue by zakajd** [photosynthesis-team/piq#147]
**Describe the bug**
In current implementation of `_compute_feats` method in `BaseFeatureMetric` class gradients are preserved, which may lead to big memory consumption. 

**Expected behavior**
Add `with torch.no_grad()` content manager to fix this issue.
